pkg/skaffold/schema/v1alpha3/config.go

```diff
@@ -98,6 +98,7 @@ type GoogleCloudBuild struct {
 // a kaniko build context
 type KanikoBuildContext struct {
 	GCSBucket string `yaml:"gcsBucket,omitempty" yamltags:"oneOf=buildContext"`
+	LocalDir  bool   `yaml:"localDir,omitempty" yamltags:"oneOf=buildContext"`
```

Comment:
I don't think using the `oneOf` tag with different types will work. IIRC the type needs to be a shared pointer type, don't think they can be primitives even if they're the same (since the "nil" type for e.g. a string isn't actually `nil`, it's `""`)